### DISCUSSION TITLE
Add Drawer Always In Top Feature.

### DISCUSSION THREAD

**Discussion by phuonglamVinaTakeuchi** [MaterialDesignInXAML/MaterialDesignInXamlToolkit#2679] [Q&A] [answered]
I used Bottom Drawer to host an Virtual Keyboard control in my app, and I want when I open the Dialog and click into an textbox in that Dialog, the bottom Drawer will be visible then I can touch the key in Virtual Keyboard to fill the textbox, but the problem is, the Dialog is always in top most, then I can not touch the Virtual Keyboard anymore. Can we have some thing way to set the Drawer alway in the top of any other object please.

**Comment by Keboo** [comment, marked as answer]
The Z-order of the DrawerHost and DialogHost can be controlled within your application. There are likely two things going on here. First, by default the `DialogHost` puts its contents inside of a Popup control, this means that the dialog itself is outside of the window and will always be on top (this has the benefit of the Dialog's content being able to be larger than the hosting window). However, if you would prefer to have the dialog contained within the window, there is a second style `MaterialDesignEmbeddedDialogHost` that you can use. This will put the dialog as part of your window's visual tree (allowing other things to be on top of it), but comes with the restriction that your dialog content cannot exceed the size of the hosting window. If it helps I have a blog post on the DialogHost [here](https://intellitect.com/material-design-in-xaml-dialog-host/)

Next, you will need to make sure that the DialogHost is contained within your DrawerHost. The order in which one of these is on-top is based on which is contained within the other, with the outer one being on top of the inner one.
